t/29_sdaf_inventory_tools.t

```diff
@@ -0,0 +1,110 @@
+use strict;
+use warnings;
+use Test::Mock::Time;
+use Test::More;
+use Test::Exception;
+use Test::Warnings;
+use Test::MockModule;
+use testapi;
+use sles4sap::sap_deployment_automation_framework::inventory_tools;
+use Data::Dumper;
+
+
+subtest '[prepare_ssh_config] ' => sub {
+    my $mock = Test::MockModule->new('sles4sap::sap_deployment_automation_framework::inventory_tools', no_auto => 1);
+    my $croak_message;
+    $mock->redefine(croak => sub { $croak_message = $_[0]; die; });
```

Comment:
I'm not sure it is a good idea to test it like this.

Tests should be about testing a behavior: I think in this case the behavior is "some argument are mandatory" 

 `croak` is about how internally you implemented this behavior. 

I suggest to try to re-implement the test not mocking croak but using  `dies_ok`